client.go

```diff
@@ -83,6 +87,21 @@ func CreateUnified() (Interface, error) {
 	return client, nil
 }
 
+// CreateUnified returns the unified client
+func CreateUnifiedWithOptions(opts *Options) (Interface, error) {
+	config, err := getRestConfigWithOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := NewClientset(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create a unified client: %v", err)
+	}
+
+	return client, nil
+}
+
 func getRestConfig() (*rest.Config, error) {
```

Comment:
I think we should eliminate getRestConfig function and leave only getRestConfigWithOptions. This function is private, so nobody from the outside uses it, and we can easily avoid code duplication here.